init.lua

```diff
@@ -1,116 +1,540 @@
 		end
 
-		local target = minetest.get_player_by_name(name_target)
-
-		if not target then
-			return true, "Unknown target player name"
+		if original_state[name_watcher] then
+			return '"' .. name_watcher .. '" is busy watching another player.'
 		end
 
-		-- avoid infinite loops
-		if original_pos[name_target] then
-			return true, name_target .. " is already watching a player."
+		if invites[name_watcher] then
+			return '"' .. name_watcher .. '" has a pending invite, try again later.'
 		end
 
-		local watcher = minetest.get_player_by_name(name_watcher)
-		local privs_watcher = minetest.get_player_privs(name_watcher)
+		if not get_player_by_name(name_watcher) then
+			return '"' .. name_watcher .. '" is not online.'
+		end
 
-		if player_api.player_attached[name_watcher] == true then
-			unwatching(name_watcher)
+		if not sm.is_permited_to_invite(name_target, name_watcher) then
+			return 'You may not invite "' .. name_watcher .. '".'
 		end
-		original_pos[name_watcher] = watcher:get_pos()
 
-		player_api.player_attached[name_watcher] = true
-		watcher:set_attach(target, "", vector.new(0, -5, -20), vector.new())
-		watcher:set_eye_offset(vector.new(0, -5, -20), vector.new())
-		watcher:set_nametag_attributes({color = {a = 0}})
+		count_invites = count_invites + 1
+		invites[name_watcher] = name_target
+		after(sm.invitation_timeout, invite_timed_out, name_watcher)
+		-- notify invited
+		chat(name_watcher, '"' .. name_target .. invitation_postfix)
 
-		toggle_hud_flags(watcher, true)
+		-- notify invitee
+		return 'You have invited "' .. name_watcher .. '".'
+	end -- invite()
 
-		watcher:set_properties({
-			visual_size = {x = 0, y = 0},
-			makes_footstep_sound = false,
-			collisionbox = {0}
-		})
+	for name_watcher in string.gmatch(param, '[^%s,]+') do
+		table.insert(messages, invite(name_watcher))
+	end
+	-- notify invitee
+	local text = table.concat(messages, '\n')
+	if 0 < count_invites then
+		text = text .. '\nThe invitations expire in '
+			.. invitation_timeout_string .. ' seconds.'
+	end
+	return true, text
+end -- watchme
 
-		privs_watcher.interact = nil
-		minetest.set_player_privs(name_watcher, privs_watcher)
 
-		return true, 'Watching "' .. name_target .. '" at '
-			.. minetest.pos_to_string(vector.round(target:get_pos()))
+-- this function only checks privs etc. Mechanics are already checked in watchme()
+-- other mods can override and extend these checks
+function spectator_mode.is_permited_to_invite(name_target, name_watcher)
+	if get_player_privs(name_target)[sm.priv_watch] then
+		return true
+	end
 
+	if not get_player_privs(name_target)[sm.priv_invite] then
+		return false
 	end
-})
 
-minetest.register_chatcommand("unwatch", {
-	description = "Unwatch a player",
-	privs = {watch=true},
-	func = function(name, param)
-		unwatching(name)
+	-- check for beerchat mute/ignore
+	local meta = get_player_by_name(name_watcher):get_meta()
+	if 'true' == meta:get_string('beerchat:muted:' .. name_target) then
```

Comment:
Is this a custom mod? Please use its API for mod compatibility for a sane dependency management. Checking against specific fields leaves no headroom for future API changes in this specific mod.

Reply:
There's `beerchat.has_player_muted_player(name, other_name)` and other benefits when using it: checking meta only works with direct mute (and likely updated in future because afaik there's no cleanup at all currently) while `beerchat.has_player_muted_player` checks registered plugins like chat jail for example so grounded players cannot bypass restrictions by spamming spectator thing.